lib/srv/db/server.go

```diff
@@ -223,8 +223,11 @@ func (s *Server) initHeartbeat(ctx context.Context, server types.DatabaseServer)
 
 func (s *Server) getServerInfoFunc(server types.DatabaseServer) func() (services.Resource, error) {
 	return func() (services.Resource, error) {
-		s.mu.Lock()
-		defer s.mu.Unlock()
+		// Make sure to return a new object, because it gets cached by
+		// heartbeat and will always compare as equal otherwise.
```

Comment:
Do you mean compare as equal by reference comparison or the actual contents?

Reply:
Yeah by reference, because heartbeat basically saves the reference to this server so any updates we make on it here would also reflect on the heartbeat's object (cause it's the same one) and so it will always compare as equal.